pkg/operators/uidgidresolver/uidgidresolver.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package uidgidresolver provides an operator that enriches events by looking
+// up uid and gid resolving them to the corresponding username and groupname.
+// Only /etc/passwd and /etc/group is read on the host. Therefore the name for a
+// corresponding could be wrong.
+package uidgidresolver
+
+import (
+	"fmt"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators/common"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	OperatorName = "UidGidResolver"
+)
+
+type UidResolverInterface interface {
+	ResolveUids() map[uint32]*string
```

Comment:
Do we envision a case where a gadget will provide multiple uids? Personally I'd try to simplify this interface to something like:

```golang
GetUid() uint32
SetUserName(string)
```

It seems to me the map is an overkill right now as almost all gadgets won't use it. Also, builtin gadgets should be deprecated soon, so let's keep these interfaces as simple as possible.

Reply:
Sure, we can do that.
I had the idea from `trace exec`, since it also provides a `loginuid` field